### PR TITLE
fix: scan mount is not unique

### LINKED ISSUE #454: Fail to create vulnerability jobs in ClientServer mode
**What steps did you take and what happened:**

Run Trivy Server in K8s
Run Trivy Operator in ClientServer mode and setting serverUrl to previously deployed Trivy Server
When Trivy Operator tries to create a vulnerability scan job for a replicaset with multiple containers it fails due to: 
"spec.template.spec.containers[1].volumeMounts[1].mountPath: Invalid value: \"/tmp/scan\": must be unique"

This does not happen when running in Standalone mode

**What did you expect to happen:**
Vulnerability scan jobs to be created and executed.

**Anything else you would like to add:**

[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Trivy-Operator version (use `trivy-operator version`):  0.1.8
- Kubernetes version (use `kubectl version`): v1.23.8
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc):

### PR DESCRIPTION
**Pull request by chen-keinan**
Signed-off-by: chenk <hen.keinan@gmail.com>

## Related issues
- Close #454 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.

### CODE REVIEW

File: pkg/plugins/trivy/plugin.go [hunk 1]
```diff
@@ -974,8 +976,6 @@ func (p *plugin) getPodSpecForClientServerMode(ctx trivyoperator.PluginContext,
 				Value: "true",
 			})
 		}
-		volumeMounts = append(volumeMounts, getScanResultVolumeMount())
-		volumes = append(volumes, getScanResultVolume())
 
 		if config.IgnoreFileExists() {
 			volumes = append(volumes, corev1.Volume{
```
Comment (JAORMX) [inline pkg/plugins/trivy/plugin.go:981]: you probably want to move this out of the loop as well.

File: pkg/plugins/trivy/plugin.go [hunk 2]
```diff
@@ -794,7 +794,9 @@ func (p *plugin) getPodSpecForClientServerMode(ctx trivyoperator.PluginContext,
 	var containers []corev1.Container
 
 	trivyConfigName := trivyoperator.GetPluginConfigMapName(Plugin)
-
+	volumeMounts = append(volumeMounts, getScanResultVolumeMount())
```
Comment (JAORMX) [inline pkg/plugins/trivy/plugin.go:797]: are all the containers using the same volume mounts?

Comment (chen-keinan) [reply pkg/plugins/trivy/plugin.go:797]: yes , probably we will have to set a different scan/.result_<num>.json name for each

File: pkg/plugins/trivy/plugin.go (hunk 1 above)
Comment (chen-keinan) [reply pkg/plugins/trivy/plugin.go:985]: you mean  the `IgnoreFileExists()` part ? it was there before, you think we should move it out ?

File: pkg/plugins/trivy/plugin.go [hunk 3]
```diff
@@ -1050,6 +1058,10 @@ func (p *plugin) getPodSpecForClientServerMode(ctx trivyoperator.PluginContext,
 	}, secrets, nil
 }
 
+func getUniqueScanResultFileName(name string) (string, error) {
```
Comment (josedonizetti) [inline pkg/plugins/trivy/plugin.go:1061]: no need to return error 
```suggestion
func getUniqueScanResultFileName(name string) string {
```

**Comment by josedonizetti** [comment]
Tested both standalone and clienserver, both working.

Standalone:
```
  - args:
    - -c
    - trivy image 'memcached' --security-checks vuln,secret --cache-dir /tmp/trivy/.cache
      --quiet --skip-update --format json > /tmp/scan/result_orion-cache.json &&  bzip2
      -c /tmp/scan/result_orion-cache.json | base64
    command:
--
  - args:
    - -c
    - trivy image 'nginx' --security-checks vuln,secret --cache-dir /tmp/trivy/.cache
      --quiet --skip-update --format json > /tmp/scan/result_orion-frontend.json &&  bzip2
      -c /tmp/scan/result_orion-frontend.json | base64
    command:
```

ClientServer:
```
  - args:
    - -c
    - trivy image 'memcached' --security-checks vuln,secret --quiet --format json
      --server 'http://my-trivy.default:4954' > /tmp/scan/result_orion-cache.json
      &&  bzip2 -c /tmp/scan/result_orion-cache.json | base64
    command:
--
  - args:
    - -c
    - trivy image 'nginx' --security-checks vuln,secret --quiet --format json --server
      'http://my-trivy.default:4954' > /tmp/scan/result_orion-frontend.json &&  bzip2
      -c /tmp/scan/result_orion-frontend.json | base64
    command:
```

File: pkg/plugins/trivy/plugin.go [hunk 4]
```diff
@@ -661,6 +661,10 @@ func (p *plugin) getPodSpecForStandaloneMode(ctx trivyoperator.PluginContext, co
 		if err != nil {
 			return corev1.PodSpec{}, nil, err
 		}
+		resultFileName := getUniqueScanResultFileName(c.Name)
+		if err != nil {
+			return corev1.PodSpec{}, nil, err
+		}
```
Comment (josedonizetti) [inline pkg/plugins/trivy/plugin.go:667]: remove

File: pkg/plugins/trivy/plugin.go [hunk 5]
```diff
@@ -1023,6 +1027,10 @@ func (p *plugin) getPodSpecForClientServerMode(ctx trivyoperator.PluginContext,
 		if err != nil {
 			return corev1.PodSpec{}, nil, err
 		}
+		resultFileName := getUniqueScanResultFileName(container.Name)
+		if err != nil {
+			return corev1.PodSpec{}, nil, err
+		}
```
Comment (josedonizetti) [inline pkg/plugins/trivy/plugin.go:1033]: remove

**Review by josedonizetti — APPROVED** [review]

**Pull request merged**